ee/desktop/runner/runner.go

```diff
@@ -213,6 +225,24 @@ func (r *DesktopUsersProcessesRunner) Execute() error {
 		case <-menuRefreshTicker.C:
 			r.refreshMenu()
 			continue
+		case <-osUpdateCheckTicker.C:
+			// on darwin, sometimes the desktop disappears after an OS update
+			// eventhough the process is still there, so lets restart desktop
+			// via killing the process and letting the runner restart it
+			if runtime.GOOS != "darwin" {
+				continue
+			}
+
+			osVersion, err := osversion()
+			if err != nil {
+				level.Error(r.logger).Log("msg", "getting os version", "err", err)
+				continue
+			}
+
+			if osVersion != r.osVersion {
+				r.osVersion = osVersion
```

Comment:
would we want some logging here to indicate why the restart happens?

Reply:
added some more logging, put in own func